tests/test_shard.py

```diff
@@ -153,15 +153,27 @@ def forward(self, data):
     torch.cuda.set_device(local_rank)
     model = Model()
 
+    def sequence_partition(module, _input):
+        world_size = dist.get_world_size()
+        sharded_seq_len = _input[0].shape[1] // world_size
+        return _input[0].split(sharded_seq_len, dim=1)[rank].contiguous()
+
     sch = slapo.create_schedule(copy.deepcopy(model))
+    sch["linear1"].sync(mode="fwd_pre", sync_op_or_fn=sequence_partition)
+    sch["linear1"].sync(mode="fwd_pre", sync_op_or_fn="all_gather", axis=1)
```

Comment:
Why we partition and all gather at the same place? Is this just for testing purpose?

Reply:
just for testing,

Reply:
It would be better to add a comment for future maintenance.